### PR TITLE
fix(file-input): validate accept entries using suffix matching

### PR DESCRIPTION
**Pull request by dtsanevmw**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

- [X] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Currently the file input validators have a regex that take all dots from the file name and taking it as the extension, for example: "file-name.0.0.1.zip" would return extension to be ".0.01.zip" while we have `accept=".zip"` and that would fail.

Issue Number: CDE-2971

## What is the new behavior?

Use suffix matching by each entry in accept so we can compare `.zip` to the file name and dynamically take only the last part of the file name.

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

File: projects/angular/src/forms/file-input/file-input-validator.ts [hunk 1]
```diff
@@ -64,4 +67,45 @@ export class ClrFileInputValidator implements Validator {
...
+        if (entry.endsWith('/*')) {
+          const prefix = entry.slice(0, entry.length - 1); // keep trailing slash
+          if (type.startsWith(prefix)) {
+            return true;
+          }
+        } else if (type === entry) {
+          return true;
+        }
+        continue;
+      }
```
Comment (valentin-mladenov) [inline projects/angular/src/forms/file-input/file-input-validator.ts:103]: Consider the following.

```suggestion
      if (entry.startsWith('.')) {
        const depth = (entry.match(/\./g) || []).length;
        if (this.getSuffixByDepth(name, depth) === entry) {
          return true;
        }
        continue;
      }

      // MIME check 
      if (entry.endsWith('/*')) {
        const prefix = entry.slice(0, entry.length - 1); // keep trailing slash
        if (type.startsWith(prefix)) {
          return true;
        }
      } else if (entry.includes('/') && type === entry) {
        return true;
      }
```

File: projects/angular/src/forms/file-input/file-input-validator.spec.ts [hunk 2]
```diff
@@ -170,6 +170,135 @@ function fileInputValidatorSpec(testComponent: Type<TestComponent>) {
...
+    fixture.detectChanges();
+    expect(nativeElement.querySelector('clr-control-error')).toBeNull();
+  });
+
+  it('empty MIME type falls back to extension checks only', () => {
+    fixture.componentInstance.accept = '.gz';
+    fixture.detectChanges();
+
+    // No MIME provided, should be accepted by extension
+    selectFiles(fileInputElement, [new File(['+'.repeat(100)], 'readme.gz', { type: '' })]);
```
Comment (valentin-mladenov) [inline projects/angular/src/forms/file-input/file-input-validator.spec.ts:302]: Add some more tests for testing in file name extensions. 

Exmaple: `readme.gz.tar` with accept `.gz` should fail.

FYI logic is working as expected just the test is missing for such a edge case.

**Review by valentin-mladenov — COMMENTED** [review]
Check my comments

File: projects/angular/src/forms/file-input/file-input-validator.ts (hunk 1 above)
Comment (dtsanevmw) [reply projects/angular/src/forms/file-input/file-input-validator.ts:103]: Fixed in [1c4489b](https://github.com/vmware-clarity/ng-clarity/pull/2014/commits/1c4489b5b6594e7565c18eb57494d2493814a84f)

File: projects/angular/src/forms/file-input/file-input-validator.spec.ts (hunk 2 above)
Comment (dtsanevmw) [reply projects/angular/src/forms/file-input/file-input-validator.spec.ts:302]: Fixed in [1c4489b](https://github.com/vmware-clarity/ng-clarity/pull/2014/commits/1c4489b5b6594e7565c18eb57494d2493814a84f)
